File: stocks/src/core/model/seq-gen.test.ts
import { SeqGen, IDGen } from "./seq-gen";

test("IDGen next generates different ids", () => {
  const sut: SeqGen = new IDGen();

  const id1 = sut.next();
  const id2 = sut.next();

  expect(id1 === id2).toBeFalsy();
});

File: core/model/seq-gen_test.go
package model

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIDGen_Next(t *testing.T) {
	var (
		gen SeqGen = IDGen{}
	)

	id1 := gen.Next()
	id2 := gen.Next()

	assert.True(t, id1 < id2)
}
